tee_supplicant/src/ree_fs.c

```diff
@@ -82,12 +82,59 @@ static int new_file(struct fs_file_t **file)
 	return -EMFILE;
 }
 
+static int internal_open(char *path, fs_mode_t flags)
+{
+	int fd, rc;
+	struct fs_file_t *file;
+
+	fd = new_file(&file);
+	if (fd < 0) {
+		rc = TEEC_ERROR_GENERIC;
+		goto free;
+	}
+
+	rc = fs_open(file, path, flags);
+	if (rc < 0) {
+		goto free;
+	}
+	return fd;
+free:
+	if (fd >= 0) {
+		remove_file(fd);
+	}
+	return rc;
+}
+
+static char *dname(char *path)
```

Comment:
What paths should handle this function? My tests show couple of strange results. Could you please comment on them?

```
Test: '/a/' result: '(null)'
Test: '/a///' result: '/a'
Test: '' result: '(null)'
Test: '/a/b/' result: '/a'
Test: 'a///b' result: 'a//'
Test: 'a/./b' result: 'a/.'
Test: 'a/./b/' result: 'a/.'
Test: 'a/../b' result: 'a/..'
Test: 'a/../b/' result: 'a/..'
Test: 'x/..' result: 'x'
```

Reply:
Investigation shows OPTEE doesn't ask to create file in sub directory. No need create directory. This patch removed